.github/workflows/windows-build.yml

```diff
@@ -0,0 +1,56 @@
+name: Windows/MinGW-W64 CI
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: windows-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+    - name: Set up MSYS2 MinGW-W64
+      uses: msys2/setup-msys2@v2
+      with:
+        msystem: mingw64
+        update: false
+        install: >-
+          git
+          zlib-devel
+          libbz2-devel
+          liblzma-devel
+          ncurses-devel
+          mingw-w64-x86_64-toolchain
+          mingw-w64-x86_64-autotools
+          mingw-w64-x86_64-tools-git
+    - name: Compile htslib
+      shell: msys2 {0}
+      run: |
+        # Todo 4th arg to track commit message of htslib#PRNUM to
+        # check out a specific htslib PR instead.
+        # Other useful vars:
+        # GITHUB_BASE_REF:  e.g. "develop"
+        # GITHUB_HEAD_REF:  e.g. "my-PR-feature" (aka GITHUB_REF_NAME?)
+        .ci_helpers/clone "https://github.com/${GITHUB_REPOSITORY_OWNER}/htslib" htslib "${GITHUB_REF_NAME}"
+        export PATH=$PATH:/mingw64/bin
+        export MSYSTEM=MINGW64
+        pushd .
+        cd htslib
+        autoreconf -i
+        ./configure
+        make -j32
```

Comment:
Configure should use `--enable-werror` so warnings will be obvious.
`make -j32` seems a bit ambitious given that the [runners only have 4 CPUs](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories).